metrics/ci_integrations/lib/source/jenkins/adapter/jenkins_source_client_adapter.dart

```diff
@@ -231,9 +235,16 @@ class JenkinsSourceClientAdapter with LoggerMixin implements SourceClient {
     return _processInteraction(interaction);
   }
 
-  /// Maps the [result] of a [JenkinsBuild] to the [BuildStatus].
-  BuildStatus _mapJenkinsBuildResult(JenkinsBuildResult result) {
-    switch (result) {
+  /// Maps the given [JenkinsBuild.building] and [JenkinsBuild.result] of the
+  /// given [build] to a [BuildStatus].
+  ///
+  /// Returns the [BuildStatus.inProgress] if the given [JenkinsBuild.building]
+  /// is `true`. Otherwise, maps the given [JenkinsBuild.result] to a
+  /// corresponding [BuildStatus].
+  BuildStatus _mapJenkinsBuildToBuildStatus(JenkinsBuild build) {
```

Comment:
Maybe we should reduce the information for this method as we want to map status only:
```suggestion
  BuildStatus _mapJenkinsResult(bool building, JenkinsResult result) {
```